src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/BookingExtractionService.kt

```diff
@@ -190,7 +192,24 @@ class BookingExtractionService(
     if (mostRecentSentencesByReleaseDate.any { it.isRecall() }) {
       dates[PRRD] = latestReleaseDate
     }
-    if (mostRecentSentencesByReleaseDate.any { !it.isRecall() }) {
+    if (booking.sentences.all { it is DetentionAndTrainingOrderSentence }) {
```

Comment:
A lot of nested if conditions in this block of code, is there any scope to lift some of the code out into methods?

Reply:
Done.